documentation/benchmark.md

```diff
@@ -0,0 +1,130 @@
+# Benchmark framework
+## Contents
+1. [Introduction](#1-introduction)
+2. [Implementation details](#2-implementation-details)
+	* [Design overview](#21-design-overview)
+	* [Timestamp source](#22-timestamp-source)
+	* [Call sequence diagram](#23-call-sequence-diagram)
+3. [Running benchmarks](#3-running-benchmarks)
+    * [Benchmark application usage](#31-benchmark-application-usage)
+    * [Adding custom timestamps](#32-adding-custom-timestamps)
+    * [Analyzing results](#33-analyzing-results)
+4. [Limitations and further steps](#4-limitations-and-further-steps)
+
+
+## 1. Introduction
+Due to its nature, OP-TEE is being a solution spanning over several
+architectural layers, where each layer includes its own complex parts.
+For further optimizations of performance, there is a need of tool which will
+provide detailed and precise profiling information for each layer.
+
+It is necessary to receive measurements of latency for:
```

Comment:
This sentence ending with each of the 2 bullets song strange not my (french) ears. Can one cross check?